frontend/app/routes/teams/index.js

```diff
@@ -19,6 +19,10 @@ export default class TeamsIndexRoute extends Route {
   afterModel(_resolvedModel, transition) {
     let team_id = transition.to.queryParams.team_id;
     let folder_id = transition.to.queryParams.folder_id;
+    let query = transition.to.queryParams.q;
+    if (query) {
```

Comment:
We don't need the `if` clause here. We want to set it no matter if it's null or not. Otherwise it would never reset